djangocms_versioning/helpers.py

```diff
@@ -43,3 +46,41 @@ def replace_admin_for_models(models, admin_site=None):
         except KeyError:
             continue
         _replace_admin_for_model(modeladmin, admin_site)
+
+
+def register_versionadmin_proxy(versionable, admin_site=None):
+    """Creates a model admin class based on `VersionAdmin` and registers
+    it with `admin_site` for `versionable.version_model_proxy`.
+
+    This model class applies filtering on the list of versions,
+    so that only versions for `versionable.content_model` are shown.
+
+    :param versionable: VersionableItem instance
+    :param admin_site: AdminSite instance
+    """
+    if admin_site is None:
+        admin_site = admin.site
+
+    if versionable.version_model_proxy in admin_site._registry:
+        # Attempting to register the proxy again is a no-op.
+        warnings.warn(
```

Comment:
Not very important but I would usually also add an assert somewhere where I'd mock the warn method and check it's being called with the message I expect.

Reply:
:+1: